### PR TITLE
fix: Update prometheus parser to be a new style parser plugin

### LINKED ISSUE #11361: Migrate prometheus to new parser structure
More info can be found [here](https://github.com/influxdata/telegraf/issues/10482)

### PR DESCRIPTION
**Pull request by reimda**
resolves #11361

### CODE REVIEW

**Comment by reimda** [comment]
There are a couple prometheus settings in config/config.go's missingTomlField func but aren't used in code. I'm not sure if they are mistakes or if they were valid settings in the past and were removed, then intentionally left in the missingTomlField so they will be ignored for backward compatibility.

* prometheus_ignore_timestamp
  - in missingTomlField
* prometheus_sort_metrics
  - in missingTomlField
  - in the prometheus serializer readme
  - in plugins/serializers/registry.go but unused elsewhere in code

I removed these in this PR but I'm having second thoughts. I need to figure out what to do with these before we merge this PR.

File: plugins/parsers/prometheus/parser.go [hunk 1]
```diff
@@ -186,3 +187,19 @@ func (p *Parser) GetTimestamp(m *dto.Metric, now time.Time) time.Time {
...
 }
+
+func (p *Parser) Init() error {
+	return nil
+}
+
+func (p *Parser) InitFromConfig(config *parsers.Config) error {
+	p.IgnoreTimestamp = config.PrometheusIgnoreTimestamp
+	return p.Init()
+}
```
Comment (srebhan) [inline plugins/parsers/prometheus/parser.go:198]: You can kill `Init` here and just return `nil`

```suggestion
func (p *Parser) InitFromConfig(config *parsers.Config) error {
	p.IgnoreTimestamp = config.PrometheusIgnoreTimestamp
	return nil
}
```

**Review by srebhan — COMMENTED** [review]
Looks good to me. Just one comment regarding `Init`.
I think your option-removal is safe, but I would prefer to do the serializer one in an extra PR to be able to easily revert it if necessary.

**Comment by reimda** [comment]
I updated this to touch only the prometheus parser and to remove the empty Init()

**Review by MyaLongmire — APPROVED** [review]

**Pull request merged**